### PR TITLE
Makefile: use git describe to assign TiDBReleaseVersion.

### PR DESCRIPTION
**Pull request by coocood**
The release version would be like `v1.0.0-69-gd122f582` if it's not exact tag, if the commit is on exact tag, it would be the tag name.

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @tiancaiamao @winkyao PTAL

File: Makefile [hunk 1]
```diff
@@ -24,6 +24,7 @@ PACKAGES  := $$(go list ./...| grep -vE "vendor")
 FILES     := $$(find . -name "*.go" | grep -vE "vendor")
 TOPDIRS   := $$(ls -d */ | grep -vE "vendor")
 
+LDFLAGS += -X "github.com/pingcap/tidb/mysql.TiDBReleaseVersion=$(shell git describe --tags --dirty)"
 LDFLAGS += -X "github.com/pingcap/tidb/util/printer.TiDBBuildTS=$(shell date -u '+%Y-%m-%d %I:%M:%S')"
 LDFLAGS += -X "github.com/pingcap/tidb/util/printer.TiDBGitHash=$(shell git rev-parse HEAD)"
 LDFLAGS += -X "github.com/pingcap/tidb/util/printer.TiDBGitBranch=$(shell git rev-parse --abbrev-ref HEAD)"
```
Comment (BusyJay) [inline Makefile]: Can these be removed?

**Comment by iamxy** [comment]
LGTM

File: Makefile (hunk 1 above)
Comment (coocood) [reply Makefile]: It is used to print branch.

**Pull request merged**